### PR TITLE
Refactored code, removed duplicate code,  vendoring it to Stork

### PR DESCRIPTION
**Pull request by gejain**
**What this PR does / why we need it**:
Refactored code, removed duplicate code,  vendoring it to Stork
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:
Result - https://aetos.pwx.purestorage.com/resultSet/testSetID/304027/testCaseID/1699261

### CODE REVIEW

**Review by Rohit-PX — APPROVED** [review]
LGTM.

**Comment by Rohit-PX** [comment]
@gejain please rebase before merging.

File: tests/common.go [hunk 1]
```diff
@@ -8691,3 +8702,71 @@ func UpdateCloudCredentialOwnership(cloudCredentialName string, cloudCredentialU
...
+	pods_migrated_len := len(pods_migrated.Items)
+	log.InfoD("Num of Pods on dest: %v", pods_migrated_len)
+	if pods_created_len != pods_migrated_len {
+		return fmt.Errorf("Pods migration failed as %v pods found on source and %v on destination", pods_created_len, pods_migrated_len)
+	}
+	destClusterConfigPath, err := GetDestinationClusterConfigPath()
+	if err != nil {
+		return err
+	}
+	err = asyncdr.ValidateCRD(appData.ExpectedCrdList, destClusterConfigPath)
```
Comment (diptiranjanpx) [inline tests/common.go:8753]: Should not we be validating the CR itself?

Comment (gejain) [reply tests/common.go:8753]: This will verify CRD seperately and we CR validation will be pods on source and destination.

Comment (diptiranjanpx) [reply tests/common.go:8753]: Its better to do the CR validation in fact, pods are just an aftereffect of the CR being present. If it cant happen in this PR, its ok but ideally we should check for the CR migration itself.

Comment (gejain) [reply tests/common.go:8753]: Sure, I will create a backlog and assign it to me for this.

**Pull request merged**